internal/kafka/internal/handlers/admin_kafka.go

```diff
@@ -223,3 +224,69 @@ func (h *adminKafkaHandler) Update(w http.ResponseWriter, r *http.Request) {
 	}
 	handlers.Handle(w, r, cfg, http.StatusOK)
 }
+
+func (h *adminKafkaHandler) validateUpdateKafkaSuspended(kafkaRequest *dbapi.KafkaRequest, kafkaUpdateReq *private.KafkaUpdateRequest, kafkaConfig *config.KafkaConfig) handlers.Validate {
+	return func() *errors.ServiceError {
+		if kafkaUpdateReq.Suspended == nil {
+			return nil
+		}
+
+		suspended := *kafkaUpdateReq.Suspended
+		if suspended {
+			err := h.validateUpdateKafkaCanBeSuspended(kafkaRequest, kafkaUpdateReq)()
+			if err != nil {
+				return err
+			}
+		} else {
+			err := h.validateUpdateKafkaCanBeResumed(kafkaRequest, kafkaUpdateReq, kafkaConfig)()
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+}
+
+func (h *adminKafkaHandler) validateUpdateKafkaCanBeSuspended(kafkaRequest *dbapi.KafkaRequest, kafkaUpdateReq *private.KafkaUpdateRequest) handlers.Validate {
+	return func() *errors.ServiceError {
+		// Kafka can only be suspended when its in a 'ready' state
+		// If Kafka is already in a 'suspending' or 'suspended' state, the request is still valid. However,
+		// no changes will be applied to the status of the Kafka instance.
+		suspendableStates := []string{constants.KafkaRequestStatusReady.String(), constants.KafkaRequestStatusSuspended.String(), constants.KafkaRequestStatusSuspending.String()}
+		isSuspendableState := arrays.AnyMatch(suspendableStates, func(status string) bool { return status == kafkaRequest.Status })
+		if !isSuspendableState {
```

Comment:
```suggestion
		if !arrays.Contains(suspendableStates, kafkaRequest.Status) {
```

Reply:
:+1: . I will keep the isSuspendableState variable however. I like to give a representative name.

Reply:
updated